src/goto-programs/goto_contractor.cpp

```diff
@@ -171,37 +178,77 @@ void goto_contractort::contractor()
   log_status("{}", oss.str());
 }
 
-ibex::CmpOp goto_contractort::get_complement(ibex::CmpOp op)
+ibex::Ctc *goto_contractort::create_contractor_from_expr2t(const expr2tc &expr)
 {
-  switch(op)
+  ibex::Ctc *contractor = nullptr;
+  auto base_object = get_base_object(expr);
+
+  if(
+    (is_comp_expr(base_object) && !is_notequal2t(base_object)) ||
```

Comment:
This if-condition looks weird to me. Can you add some comments?

Reply:
Added to the documentation of this function.